client/my-sites/plugins/plugin.jsx

```diff
@@ -113,9 +116,8 @@ export default React.createClass( {
 
 	refreshSitesAndPlugins( nextProps ) {
 		this.setState( this.getSitesPlugin( nextProps ) );
-
 		// setTimeout to avoid React dispatch conflicts.
-		setTimeout( () => {
+		this.pluginRefreshTimeout = setTimeout( () => {
```

Comment:
The problem is due that timeout, the title change code is being executed even if this component has been unmounted already ... that's why this component was overwriting the title even if this component was already dead.
With this second internal timeout, we can clear it and not launch the title edit code when the component is unmounted, avoiding nasty side effects! :)